rpc/metrics.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2020 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package rpc
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/metrics"
+)
+
+var (
+	jsonrpcRequestGauge    = metrics.NewRegisteredGauge("rpc/jsonrpc/count/in", nil)
+	successfulRequestGauge = metrics.NewRegisteredGauge("rpc/jsonrpc/count/success", nil)
+	failedReqeustGauge     = metrics.NewRegisteredGauge("rpc/jsonrpc/count/failure", nil)
```

Comment:
I'd drop the `count` from these 3 metrics. Maybe change the `in` to `requests` and then everything is fairly obvious. Since these are gouges, the `.count` will be exposed in the metrics gathering apis anyway automatically.